src/main/java/io/appform/jsonrules/expressions/array/CollectionJsonPathBasedExpression.java

```diff
@@ -60,13 +62,13 @@ protected CollectionJsonPathBasedExpression(ExpressionType type, String path, @S
     @Override
     protected final boolean evaluate(ExpressionEvaluationContext context, String path, JsonNode evaluatedNode) {
         if (extractValues) {
-            JsonNode jsonNode = mapper.valueToTree(JsonPath.using(ComparisonUtils.SUPPRESS_EXCEPTION_CONFIG)
-                    .parse(context.getNode().toString()).read(String.valueOf(valuesPath)));
+            JsonNode jsonNode = JsonPath.using(JsonUtils.SUPPRESS_EXCEPTION_CONFIG)
+                    .parse(context.getNode()).read(String.valueOf(valuesPath));
             if (jsonNode == null || !jsonNode.isArray()) {
                 return false;
             }
             // fetch values from @values path as a set.
-            final HashSet<Object> extractedPathValues = new HashSet<>(JsonPath.read(jsonNode.toString(), "$"));
+            final Set<Object> extractedPathValues = new HashSet<>(mapper.convertValue(jsonNode, new TypeReference<Collection<Object>>() {}));
```

Comment:
Why this create is neeeded? We can't send `TypeReference<Set<Object>>` directly?